# Generated by Django 3.1.5 on 2021-01-21 15:29

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ('accounts', '0003_auto_20210121_2039'),
    ]

    operations = [
        migrations.CreateModel(
            name='CustomUserFollowing',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('created', models.DateTimeField(auto_now_add=True, db_index=True)),
                ('follower', models.ForeignKey(help_text='The user that is doing the following', on_delete=django.db.models.deletion.CASCADE, related_name='followers', to=settings.AUTH_USER_MODEL)),
                ('user', models.ForeignKey(help_text='The user that is being followed', on_delete=django.db.models.deletion.CASCADE, related_name='following', to=settings.AUTH_USER_MODEL)),
            ],
            options={
                'ordering': ['-created'],
            },
        ),
        migrations.DeleteModel(
            name='UserFollowing',
        ),
        migrations.AddConstraint(
            model_name='customuserfollowing',
            constraint=models.UniqueConstraint(fields=('user', 'follower'), name='unique_followers'),
        ),
    ]
